run_kraken.py

```diff
@@ -12,6 +12,88 @@
+        cerberus_url = config["cerberus"]["cerberus_url"]
+        if not cerberus_url:
+            logging.error("url where Cerberus publishes True/False signal "
+                          "is not provided.")
+            sys.exit(1)
+        cerberus_status = requests.get(cerberus_url).content
+        cerberus_status = True if cerberus_status == b'True' else False
+        if not cerberus_status:
+            logging.error("Received a no-go signal from Cerberus, looks like the"
+                          " cluster is unhealthy. Please check the Cerberus report"
+                          " for more details. Test failed.")
+            if failed_post_scenarios:
+                if config['kraken']['exit_on_failure']:
+                    logging.info("Cerberus status is not healthy and post action scenarios "
+                                 "are still failing, exiting kraken run")
+                    sys.exit(1)
+                else:
+                    logging.info("Cerberus status is not healthy and post action scenarios "
+                                 "are still failing")
+        else:
+            logging.info("Received a go signal from Ceberus, the cluster is "
+                         "healthy. Test passed.")
+
+            if failed_post_scenarios:
+                if config['kraken']['exit_on_failure']:
+                    logging.info("Cerberus status is healthy but post action scenarios "
+                                 "are still failing, exiting kraken run")
+                    sys.exit(1)
+                else:
+                    logging.info("Cerberus status is healthy but post action scenarios "
+                                 "are still failing")
+    else:
+        if config['kraken']['exit_on_failure']:
+            if failed_post_scenarios:
+                logging.info("Post action scenarios are failing, exiting kraken run")
+                sys.exit(1)
+
+
+# Perform the post scenario actions to see if components recovered
+def post_actions(kubeconfig_path, scenario, failed_post_scenarios):
+    for failed_scenario in failed_post_scenarios:
+        post_action_output = runcommand.invoke("powerfulseal autonomous "
+                                               "--use-pod-delete-instead-of-ssh-kill"
+                                               " --policy-file %s --kubeconfig %s --no-cloud"
+                                               " --inventory-kubernetes --headless"
+                                               % (failed_scenario, kubeconfig_path))
+        if "ERROR" not in post_action_output:
+            failed_post_scenarios.remove(failed_scenario)
+        else:
+            logging.info('Post action scenario ' + str(failed_scenario) + "is still failing")
+            error_message = post_action_output.split("ERROR")[1].split('\n')[0]
+            logging.info("Error message was " + str(error_message))
+    # check post actions
+    if len(scenario) > 1:
```

Comment:
I think it would be nice to allow the user to include multiple post_actions for each pod_scenario. Thoughts?

Reply:
I think it would be nice to be able to do multiple checks I'm just not sure how often it would be used. I think the post action pod scenarios and writing your own custom script could cover more than one check in itself and return a failure if any part of it faills. @mffiedler @chaitanyaenr what are your thoughts on this?

Reply:
+1, think the custom script option should cover scenarios having multiple checks and we can also mention a list of actions in the same config in case of multiple things to check.